web/src/api/helpers.ts

```diff
@@ -0,0 +1,58 @@
+import invariant from 'tiny-invariant';
+
+async function sha256(message: string): Promise<string> {
+  const BASE = 16;
+  const MAX_LENGTH = 2;
+  const hashBuffer = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(message));
+  return [...new Uint8Array(hashBuffer)].map((b) => b.toString(BASE).padStart(MAX_LENGTH, '0')).join('');
+}
+
+/**
+ * Use local endpoint only if ALL of the following conditions are true:
+1. The app is running on localhost
+2. The `remote` search param hasn't been explicitly set to true
+3. Document domain has a non-empty value
+ * @returns
+ */
+function isUsingLocalEndpoint(): boolean {
+  return import.meta.env.DEV && window.location.href.includes('remote=true') && document.domain !== '';
```

Comment:
Minor, but important change 😉 
```suggestion
  return import.meta.env.DEV && !window.location.href.includes('remote=true') && document.domain !== '';
```
Also, VSCode says `document.domain` is deprecated